jdaviz/configs/imviz/plugins/viewers.py

```diff
@@ -171,3 +185,46 @@ def data(self, cls=None):
                 for layer_state in self.state.layers
                 if hasattr(layer_state, 'layer') and
                 layer_is_image_data(layer_state.layer)]
+
+    def get_link_type(self, data_label):
+        """Find the type of ``glue`` linking between the given
+        data label with the reference data in viewer.
+
+        Parameters
+        ----------
+        data_label : str
+            Data label to look up.
+
+        Returns
+        -------
+        link_type : {'pixels', 'wcs', 'self'}
+            One of the link types accepted by :func:`~jdaviz.configs.imviz.helper.link_image_data`
+            or ``'self'`` if the data label belongs to the reference data itself.
+
+        Raises
+        ------
+        ValueError
+            Link look-up failed.
+
+        """
+        if self.state.reference_data is None:
+            raise ValueError('No reference data for link look-up')
+
+        ref_label = self.state.reference_data.label
+        if data_label == ref_label:
+            return 'self'
+
+        link_type = None
+        for elink in self.session.application.data_collection.external_links:
+            elink_labels = (elink.data1.label, elink.data2.label)
+            if data_label in elink_labels and ref_label in elink_labels:
+                if isinstance(elink, LinkSame):  # Assumes WCS link never uses LinkSame
+                    link_type = 'pixels'
+                else:  # If not pixels, must be WCS
+                    link_type = 'wcs'
+                break  # Might have duplicate, just grab first match
```

Comment:
In what scenario would this happen (and would the "duplicate" ever be something different than the first match)?

Reply:
In this scenario:

```python
>>> import numpy as np
>>> from jdaviz import Imviz
>>> a = np.ones((10, 10))
>>> imviz = Imviz()
>>> imviz.load_data(a, data_label='1')
>>> imviz.load_data(a, data_label='2')
>>> imviz.load_data(a, data_label='3')
>>> for elink in imviz.app.data_collection.external_links:
...     print(elink.__class__.__name__, elink.data1.label, elink.data2.label)
LinkSame 1 2
LinkSame 1 2
LinkSame 1 3
LinkSame 1 3
```

Figuring out why is a rabbit hole for another day.

Reply:
So far in my limited testing, the duplicate is always the same.

There is also another oddity I encountered -- When I load pure numpy arrays, it takes one extra blink for the app to report the correct visible layer data label. Again, "future work."

Reply:
ok, as long as we would never expect the scenario where the first case gives `LinkSame -> pixels` but another (ignored because of the break) case would give `wcs`, then I think this is probably fine.